api/v1alpha1/edgedevice_types.go

```diff
@@ -36,6 +36,16 @@ type EdgeDeviceSpec struct {
 
 	Heartbeat *HeartbeatConfiguration `json:"heartbeat,omitempty"`
 	Storage   *Storage                `json:"storage,omitempty"`
+	Metrics   *MetricsConfiguration   `json:"metrics,omitempty"`
+}
+
+type MetricsConfiguration struct {
+	Retention *Retention `json:"retention,omitempty"`
+}
+
+type Retention struct {
+	MaxMiB   int32 `json:"maxMiB"`
```

Comment:
I would add here the description and the kubebuilder minimum if it's set.

Reply:
Done.